schemas/com.google.tag-manager.server-side/generate_lead/jsonschema/1-0-0

```diff
@@ -0,0 +1,24 @@
+{
+        "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+        "self": {
+                "vendor": "com.google.tag-manager.server-side",
+                "name": "generate_lead",
+                "version": "1-0-0",
+                "format": "jsonschema"
+        },
+        "description": "A generate_lead GTM server-side event.",
+        "type": "object",
+        "properties": {
+                "currency": {
+                        "description": "Currency of the items associated with the event, in 3-letter ISO 4217 format. Is required, if value is set.",
```

Comment:
There's `dependentRequired` to express this dependency. Not very popular in Iglu Central, but would send an event into bad bucket during enrich https://json-schema.org/understanding-json-schema/reference/conditionals.html

Reply:
Concerning `dependentRequired`, the GA event seems to go through (at least until server-side tag) and be handled, but the Google documentation seems sparse. So, maybe we could allow it less strict here, even though not recommended?

The last force-pushed change to mention, is `items` that are changed to an `oneOf`, to not invalidate events that are still valid according to google docs. (required is either `item_id` or `item_name`).